### ISSUE TITLE
Macro link scrolls the whole page down to a wrong position

### ISSUE THREAD

**Issue by OleksandrKvl** [doxygen/doxygen#11285]
**Describe the bug**
Hi, I recently updated my project documentation from Doxygen 1.10 to 1.12 and noticed one strange issue. When I click on a macro link, instead of pointing me to that macro documentation, it navigates me to the page where macro is documented and scrolls it down. Each time I click on macro link, it scrolls lower and lower.

**To Reproduce**
You can observe this behavior on my project's page, documentation for version 1.4.2 was generated using Doxygen 1.10 and behaves correctly while docs for version 1.5.0 was generated using Doxygen 1.12.
Go here: https://oleksandrkvl.github.io/sbepp/1.5.0/representation.html#autotoc_md20. At the bottom of "Safety checks" section there are links to various `SBEPP_abc` macros, let's take [SBEPP_DISABLE_ASSERTS](https://oleksandrkvl.github.io/sbepp/1.5.0/sbepp_8hpp.html#a6a9a3e2c289e8eb83bd1937c0a54b14e) for example. When I click on it, I'm redirected to the correct page but a wrong position on that page. Similar [SBEPP_DISABLE_ASSERTS](https://oleksandrkvl.github.io/sbepp/1.4.2/sbepp_8hpp.html#a6a9a3e2c289e8eb83bd1937c0a54b14e) link generated by 1.10 works as expected.
When you open `SBEPP_DISABLE_ASSERTS` documentation, on the left there's a bar with other names, if you continue to press on `SBEPP_DISABLE_ASSERTS` link there, page will scroll down more and more.

My docs use custom CSS but I observed the same behavior locally without it.

**Expected behavior**
Link should point to a correct page position.

**Version**
Doxygen 1.12
Firefox 133.0.3

**Comment by OleksandrKvl** [comment]
Here's a screen capture demo, it shows different behavior between versions 1.5.0 (Doxygen 1.12) and 1.4.2 (Doxygen 1.10).
![Screen Recording 2024-12-19 at 18 49 56](https://github.com/user-attachments/assets/e5d877b7-59ed-4ff7-8ac9-462d75a04feb)

@albert-github I can try to create something simple but I think you already have such simple examples of documented macros so I'm not sure it will help or even reproduce the issue.

**Comment by albert-github** [comment]
I saw the github io page and I see the difference between tour 1.4,2 with doxygen 1.10.0 versions and your 1.5.0 with doxygen 1.12.0 but to be able to do some tests and to see where the problem lies it is necessary to have a "small" example or at least a pointer to the repository (with instructions how to build the documentation)

**Comment by OleksandrKvl** [comment]
OK, for now it's faster for me to give you build instructions rather trying to reproduce it with a small example. It's pretty simple: clone https://github.com/OleksandrKvl/sbepp and run these commands:

```bash
# CMake configuration step
cmake -DSBEPP_BUILD_SBEPPC=OFF -DSBEPP_BUILD_DOCS=ON -B $BUILD_DIR
# build the docs
cmake --build $BUILD_DIR --target doc
```

Replace `$BUILD_DIR` with the name you like. After the build step docs will be in `$BUILD_DIR/doc/docs/1.5.0` directory. It uses Doxygen detected by `find_package(Doxygen REQUIRED)`. Let me know if that works for you, if it doesn't, I will try to create a small demo tomorrow.

**Comment by OleksandrKvl** [comment]
Not sure if this helps (I'm not a front-end developer) but if I comment `docContent.animate()` call in `navtree.js:gotoAnchor()`, this weird scrolling down goes away. Also I've noticed that although in 1.10 click leads to a correct page position, it first scrolls the page down a bit and then jumps back to a correct position. That scroll is not needed, it moves page to a wrong position only to then sharply jump back to a correct one. And in 1.12 it only does the scroll part without the final jump.

**Comment by albert-github** [comment]
I have to check, but the information you provided might be useful and did ring a bell.
I did build the 1.5.o version of sbepp with doxygen 1.10.0 and 1.12.0 version and saw the problem you describe. With the current master (1.13.0 (163ed03282c894bd9a15cc2a81182cb5336daefd)) it looks to me that the problem is gone.

I "started" in all 3 cases the `index.html` and in the search box I entered `SBEPP_DISABLE_ASSERTS` in clicked on the item shown.

I think the problem is fixed but not released.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by OleksandrKvl** [comment]
Works as expected with 1.13, thank you.
